### PR TITLE
Add save story error message

### LINKED ISSUE #801: Show snackbar message when story fails to save
## Feature Description
Show an error message when if the story fails to save for whatever reason. 

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by merapi**
Fixes #801 

However, I don't know if this is enough for such critical action.

On FHD it's barely noticeable, given the distance between button and message:
image

### CODE REVIEW

**Comment by swissspidy** [comment]
Doesn't the snackbar provider support various positions? cc @ndev1991 

Would need UX confirmation on best position

**Comment by merapi** [comment]
> Doesn't the snackbar provider support various positions? cc @ndev1991

It does, but only at Provider level (cannot be overridden for a specific snackbar), so this should be changed IMHO.

```jsx
<SnackbarProvider place="top-right">
```
CC @samitron7 for UX

image

**Comment by spacedmonkey** [comment]
The code for this looks good ( it is very simple ). Once UX has been confirmed, I will approve.

**Comment by samitron7** [comment]
The Material recommendation is for snackbar to always be at the bottom lower left corner. It's definitely not ideal and we're looking slightly changing up the notification system in P1 but in the meantime, I would go with the Material recommendation. 


Screen Shot 2020-03-31 at 2 09 44 PM

**Comment by merapi** [comment]
@spacedmonkey Let's merge it for Alpha.

**Review by spacedmonkey — APPROVED** [review]
LGTM

**Comment by spacedmonkey** [comment]
@merapi Approved

File: assets/src/edit-story/app/story/actions/useSaveStory.js [hunk 1]
```diff
@@ -125,7 +132,9 @@ function useSaveStory({ storyId, pages, story, updateStory }) {
         refreshPostEditURL(storyId);
       })
       .catch(() => {
-        // TODO Display error message to user as save as failed.
+        showSnackbar({
+          message: __('Failed to save the story', 'web-stories'),
```
Comment (barklund) [inline assets/src/edit-story/app/story/actions/useSaveStory.js:136]: Should we include some extra technical information? If a superuser wants to try to do something about it, they need more info. Or if any of our alpha/beta testers come across this, they need more information in order to file a better bug report.

Can we add extra info to the snack bar, that can be expanded, or something like that? Something like a _"Show technical information"_ toggle?

**Review by barklund — COMMENTED** [review]
I'd like us to give better feedback, in order for users of the plugin to file better bug reports.

File: assets/src/edit-story/app/story/actions/useSaveStory.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/story/actions/useSaveStory.js:136]: What like this https://github.com/WordPress/gutenberg/pull/15657 ?

Comment (spacedmonkey) [reply assets/src/edit-story/app/story/actions/useSaveStory.js:136]: I also think that we can follow up on this after.

**Review by dvoytenko — APPROVED** [review]
LGTM. Would also like to see a follow up with some user-actionable messaging. Such as "Try again.". Assuming we can tell the reason for why saving fails.

**Comment by swissspidy** [comment]
@jauyong Can you create a follow-up task for more actionable messaging here?

**Pull request merged**